operators/constellation-node-operator/internal/cloud/azure/client/nodeimage.go

```diff
@@ -27,12 +28,34 @@ func (c *Client) GetNodeImage(ctx context.Context, providerID string) (string, e
 	if resp.Properties == nil ||
 		resp.Properties.StorageProfile == nil ||
 		resp.Properties.StorageProfile.ImageReference == nil ||
-		resp.Properties.StorageProfile.ImageReference.ID == nil && resp.Properties.StorageProfile.ImageReference.CommunityGalleryImageID == nil {
+		resp.Properties.StorageProfile.ImageReference.ID == nil &&
+			resp.Properties.StorageProfile.ImageReference.CommunityGalleryImageID == nil &&
+			resp.Properties.StorageProfile.ImageReference.Publisher == nil &&
+			resp.Properties.StorageProfile.ImageReference.Offer == nil &&
+			resp.Properties.StorageProfile.ImageReference.SKU == nil &&
+			resp.Properties.StorageProfile.ImageReference.Version == nil {
```

Comment:
```suggestion
		resp.Properties.StorageProfile.ImageReference.ID == nil &&
			resp.Properties.StorageProfile.ImageReference.CommunityGalleryImageID == nil &&
			(resp.Properties.StorageProfile.ImageReference.Publisher == nil ||
			resp.Properties.StorageProfile.ImageReference.Offer == nil ||
			resp.Properties.StorageProfile.ImageReference.SKU == nil ||
			resp.Properties.StorageProfile.ImageReference.Version == nil) {
```
Shouldn't this use "or" instructions since we want to throw an error if `ImageReference.ID`, `ImageReference.CommunityGalleryID` and any of the marketplace values are nil?